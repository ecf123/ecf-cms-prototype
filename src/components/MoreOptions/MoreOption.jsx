import React from "react";
import { ReactComponent as BinIcon }from "../../assets/bin-icon.svg";
import { ReactComponent as PenIcon }from "../../assets/pen-icon.svg";
import './MoreOptions.scss';
const MoreOption = (props) => {
    const {item,description} = props;
  return (
    <div className="more-options">
      <h3 className="more-options__title">{item} Options</h3>
      <p className="more-options__description">{description}</p>
      <div className="more-options__buttons">
        <button className="more-options__button"><PenIcon></PenIcon>Edit</button>
        <button className="more-options__button more-options__button--delete"><BinIcon></BinIcon>Delete</button>
      </div>
    </div>
  );
};

export default MoreOption;
