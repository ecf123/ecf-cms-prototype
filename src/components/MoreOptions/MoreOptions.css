@import url("https://fonts.googleapis.com/css2?family=Inter:wght@400;500&family=Lexend+Deca:wght@400;700&display=swap");
* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

html {
  font-size: 100% !important;
}

* {
  font-family: "Lexend Deca", sans-serif;
  color: #f3f2e8;
}

p {
  margin: 0.5rem 0;
  display: inline-block;
  width: 100%;
}

button {
  font-size: 1rem;
}

h1 {
  font-size: 2.5rem;
  font-weight: 400;
}

h2 {
  font-size: 2rem;
  font-weight: 700;
}

h3 {
  font-size: 1.75rem;
  font-weight: 700;
}

h4 {
  font-size: 1.5rem;
  font-weight: 400;
}

h5 {
  font-size: 1rem;
  font-weight: 400;
}

h6 {
  font-size: 0.8rem;
  font-weight: 400;
}

@media screen and (min-width: 768px) {
  :root {
    font-size: 110%;
  }
}
@media screen and (min-width: 1024px) {
  :root {
    font-size: 120%;
  }
}
.more-options {
  background-color: #ffffff;
  width: 23.1rem;
  padding: 1.25rem;
  padding-right: 2.5rem;
  border-radius: 0.63rem;
  border: 0.06rem solid #f5f5f5;
}
.more-options__buttons {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: 12rem;
}
.more-options__button {
  padding: 0.63rem 1.25rem;
  margin: 0.32rem;
}
.more-options__title {
  color: #2d303a;
  font-size: 1rem;
  margin: 0.63rem;
  font-family: "Inter", sans-serif;
}
.more-options__description {
  color: #5d7582;
  font-size: 0.8rem;
  margin: 0.63rem;
  font-family: "Inter", sans-serif;
}
