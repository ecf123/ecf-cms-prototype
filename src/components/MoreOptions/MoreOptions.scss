@use "../../styles/variables" as *;
@use "../../styles/main.scss" as *;

.more-options {
  background-color: $color-white;
  padding: 1.25rem;
  padding-right: 2.5rem;
  border-radius: 0.63rem;
  border: 0.06rem solid $color-grey-light;
  &__buttons {
    display: flex;
    flex-direction: row;
  }
  &__button {
    padding: 0.63rem 1.25rem;
    margin: 0.32rem;
  }
  &__title {
    color: $color-blue-navy-dark;
    font-size: $font-size-small;
    margin: 0.63rem;
    font-family: $font-secondary;
  }
  &__description {
    color: $color-teal;
    font-size: $font-size-extra-small;
    margin: 0.63rem;
    font-family: $font-secondary;
  }
}
