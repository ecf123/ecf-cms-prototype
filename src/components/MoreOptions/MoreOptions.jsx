import "./MoreOptions.scss";
import Button from "../Button/Button";
import { ReactComponent as Bin } from "../../assets/bin-icon.svg";
import { ReactComponent as Pencil } from "../../assets/pen-icon.svg";

const MoreOptions = ({ title, description, handleDelete, handleEdit }) => {
  return (
    <div className="more-options">
      <h3 className="more-options__title">{title} Options</h3>
      <p className="more-options__description">{description}</p>
      <div className="more-options__buttons">
        <Button
          onClick={handleEdit}
          style={"button light-grey round-border medium"}
          imgStyle={"medium-img black-svg"}
          textStyle={"text medium-text light-grey"}
          buttonText={"Edit"}
        >
          <Pencil />
        </Button>
        <Button
          onClick={handleDelete}
          style={"button red round-border medium"}
          imgStyle={"medium-img white-svg"}
          textStyle={"text medium-text red"}
          buttonText={"Delete"}
        >
          <Bin />
        </Button>
      </div>
    </div>
  );
};

export default MoreOptions;
