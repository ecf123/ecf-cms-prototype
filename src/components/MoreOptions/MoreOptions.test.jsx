import { customRender } from "../../utils/testUtils";
import MoreOptions from "./MoreOptions";

it("Should render MoreOption component", () => {
  const { container } = customRender(<MoreOptions title={"More"} />);

  expect(container).toMatchSnapshot();
});
