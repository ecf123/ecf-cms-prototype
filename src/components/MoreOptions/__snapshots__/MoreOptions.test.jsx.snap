// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MoreOption component 1`] = `
<div>
  <div
    class="more-options"
  >
    <h3
      class="more-options__title"
    >
      More
       Options
    </h3>
    <p
      class="more-options__description"
    />
    <div
      class="more-options__buttons"
    >
      <button
        class="button grey round-border small more-options__button"
      >
        <div
          class="small-img black-svg"
        >
          <svg>
            pen-icon.svg
          </svg>
        </div>
        <p
          class="text small-text grey"
        >
           
          Edit
        </p>
      </button>
      <button
        class="button red round-border small more-options__button"
      >
        <div
          class="small-img white-svg"
        >
          <svg>
            bin-icon.svg
          </svg>
        </div>
        <p
          class="text small-text red"
        >
           
          Delete
        </p>
      </button>
    </div>
  </div>
</div>
`;
